CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -331,27 +331,36 @@ public static function remove($groupID = NULL, $onceOnly = TRUE) {
 ";
       }
       else {
-
-        $query = "
-DELETE     gc
-FROM       civicrm_group_contact_cache gc
-INNER JOIN civicrm_group g ON g.id = gc.group_id
-WHERE      g.cache_date <= %1
-";
+        $cacheTime = date('Y-m-d H-i-s', strtotime("- $smartGroupCacheTimeout minutes"));
+        $params = array(1 => array($cacheTime, 'String'));
+        $groupsQuery = "SELECT id
+          FROM civicrm_group
+          WHERE g.cache_date <= %1";
+        $dao = CRM_Core_DAO::executeQuery($groupsQuery, $params);
+        $groupsIDs = array();
+        while ($dao->fetch()) {
+          $groupIds[] = $dao->id;
+          $query = "
+             DELETE
+             FROM civicrm_group_contact_cache
+             WHERE group_id = %1";
+          $queryParams = array(1 => array($dao->id, 'Integer'));
+          CRM_Core_DAO::executeQuery($query, $queryParams);
+        }
+        $groups = implode(', ', $groupIds);
         $update = "
-UPDATE civicrm_group g
-SET    cache_date = null,
+UPDATE civicrm_group
+SET    cache_date = $now,
```

Comment:
OK, my comment seems to be wrong. `$now` is technically numeric (YmdHis), and MySQL CLI accepts `cache_date` in that format, so maybe it doesn't need escaping.